cli/src/main.rs

```diff
@@ -24,64 +24,85 @@ use solana_sdk::{
     transaction::Transaction,
 };
 use spl_associated_token_account::{create_associated_token_account, get_associated_token_address};
-use spl_token::state::Mint;
+use spl_token::state::{Account, Mint};
 
 struct Config {
     owner: Box<dyn Signer>,
     fee_payer: Box<dyn Signer>,
     rpc_client: RpcClient,
 }
 
-fn eth_pubkey_of(matches: &ArgMatches<'_>, name: &str) -> Result<secp256k1::PublicKey> {
-    let mut value = value_t!(matches.value_of(name), String)?;
-    // handle string with front-going hex prefix
-    if value.len() == secp256k1::util::FULL_PUBLIC_KEY_SIZE + 2 && value.starts_with("0x") {
-        value.replace_range(..=1, "");
+fn handle_hex_prefix(str: &mut String) {
+    if str.starts_with("0x") {
+        str.replace_range(..=1, "");
     }
+}
+
+fn eth_pubkey_of(matches: &ArgMatches<'_>, name: &str) -> anyhow::Result<secp256k1::PublicKey> {
+    let mut value = value_t!(matches.value_of(name), String)?;
+    handle_hex_prefix(&mut value);
     let decoded_pk = &hex::decode(value.as_str())?;
     let pk = secp256k1::PublicKey::parse_slice(decoded_pk.as_slice(), None)?;
     Ok(pk)
 }
 
-fn eth_seckey_of(matches: &ArgMatches<'_>, name: &str) -> Result<secp256k1::SecretKey> {
+fn eth_seckey_of(matches: &ArgMatches<'_>, name: &str) -> anyhow::Result<secp256k1::SecretKey> {
```

Comment:
I'm not against using new crates (especially here off-chain), but in the future let's talk first before adding any dependencies that affect code style.